.github/workflows/ci.yml

```diff
@@ -182,28 +193,44 @@ jobs:
           SIGNING_KEY: ${{ secrets.SIGNING_KEY }}
         run: .github/ci.sh sign $NAME-with-solvers.tar.gz
 
-      - if: matrix.ghc == '8.10.7'
+      - if: matrix.ghc == '8.10.7' && matrix.hpc == false
         uses: actions/upload-artifact@v2
```

Comment:
Why are we explicitly checking `matrix.hpc == false` here?  If the ghc version is because this is (arbitrarily) our current "distribution" version, won't this break if we then happen to choose the distribution version to be the same one we are generating coverage for?  Same question goes for the next 2 steps with the same check.  Either something to fix, or something that should have an explanatory comment.

Reply:
The check is here to avoid the problem you identified. If the distribution version is changed to match the HPC version, then this prevents the HPC binaries from clobbering the distribution binaries. For example, assume the distribution is changed to `9.4.4`. The matrix specifies that there are two different Ubuntu builds with GHC 9.4.4. One of those builds has HPC enabled and the other has HPC disabled. This conditional ensures that it's the version with HPC disabled that gets uploaded.  I'll add a comment to clarify that.

Reply:
I added a comment clarifying the `matrix.hpc == false` check

Reply:
Let me know if you still think there's a logic error here